pwnlib/tubes/ssh.py

```diff
@@ -1074,15 +1074,16 @@ def is_exe(path):
                 python = ssh_process(self, script, tty=True, cwd=cwd, raw=True, level=self.level, timeout=timeout)
 
             try:
-                python.recvline_contains(b'PWNTOOLS')        # Magic flag so that any sh/bash initialization errors are swallowed
-                python.recvline()                           # Python interpreter that was selected
+                python.recvline_contains(b'PWNTOOLS')   # Magic flag so that any sh/bash initialization errors are swallowed
+                if b'python' not in python.recvline():  # Python interpreter that was selected
+                    self.warn_once('Could not find a Python interpreter on %s\n' % self.host
+                                   + "Use ssh.system() instead of ssh.process()\n")
+                    h.failure("Process creation failed")
+                    return None
+
                 result = safeeval.const(python.recvline())  # Status flag from the Python script
             except (EOFError, ValueError):
                 h.failure("Process creation failed")
-                self.warn_once('Could not find a Python interpreter on %s\n' % self.host
```

Comment:
This error might still be useful on EOF, maybe just strip the recvall call, but keep it as well.

Reply:
Thanks! Put the python check in it's own catch block so it warns on eof from this `recvline()` call. The problem before was that the proceeding `safeval.const()` call failed, but I recieved a warning message that no python could be found on the server, even though there was one.